machineconfiguration/v1/types.go

```diff
@@ -745,3 +749,169 @@ type ContainerRuntimeConfigList struct {
 
 	Items []ContainerRuntimeConfig `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineState describes the health of the Machines on the system
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
```

Comment:
born into compat-level 1 is very confident.  Are we already that confident that we are getting these types right, and won't need tough follow-up changes?

Reply:
not sure what the protocol is here. I was under the impression that anything user facing should be level 1?

Reply:
I am planning on featuregating, do you think I need to introduce that in this PR?

Reply:
We can do the feature gate as a separate PR, that's a pretty easy thing to add so can probably get that in and merged before the API